airflow/jobs.py

```diff
@@ -738,8 +749,20 @@ def _execute(self):
                         dag = dagbag.dags[ti_key[0]]
                         task = dag.get_task(ti_key[1])
                         ti = TI(task, ti_key[2])
+                        ti.refresh_from_db(session=session, lock_for_update=True)
+                        if ti.state == State.SCHEDULED:
+                            session.commit()
+                            self.logger.debug("Task {} was picked up by another scheduler"
```

Comment:
I don't think we support multiple schedulers officially? Without locks, there a lot of race conditions.